hw/ip/spi_host/doc/theory_of_operation.md

```diff
@@ -1,87 +1,71 @@
+The host is responsible for generating the serial clock, and typically each side asserts outgoing data on one edge of the clock (e.g. on the rising edge) and samples data in the next edge (e.g. on the falling edge).
+The SPI bus can be connected to many target devices, but each device on the bus must get its own chip select, and so this signal designates a particular device for each command.
+In this IP we refer to the chip select as `CSB`.
+
+Bus traffic is composed of 'commands', each of which is initiated by the host and targets a single device.
+The assertion and deassertion of the target devices's chip-select marks the beginning and end of the command.
+No device will accept any command input until `CSB` has been asserted for that device, and most devices (if not all) do not accept a second command unless `CSB` has been deasserted to mark the end of the previous command.
+Some simple devices, particularly those that support SPI daisy-chaining, do not process command input at all until *after* the `CSB` line has been deasserted.
+In the case of NOR flash devices, read and write commands are indeterminate in length, and the data transfer ends only when the `CSB` line is deasserted.
+So, although the exact details of the operation may vary from device to device, the edges of the `CSB` signal always serve as important markers for delineating the boundaries of each transaction.
+
+Typical SPI commands are categorized as Standard, Dual, or Quad depending on the number of serial data lines (`SD[x]`) they utilize.
+This is often referred to as the 'Speed-Mode' or just 'Speed', although as the clock frequency remains constant this is a slight misnomor, so it may be better thought-of as a 'Throughput-Mode' or just a data width.
+A command may be composed of a number of phases or segments, where different numbers of data lines are used in different directions.
+The OpenTitan SPI_HOST IP uses the term '[segment](#command-interface)' to describe this smaller unit of traffic, and each segment can have it's own data direction and speed-mode.
+For example, in a Standard SPI command a number of instruction-, address- or data-bytes are transmitted on `SD[0]`, and response bytes are received on `SD[1]`.
+Therefore, for Standard-mode commands `SD[0]` is always configured as an output and `SD[1]` is always an input.
+In this mode, the bus can be effectively used as a full-duplex communication channel.
+Not all devices exploit this capability however, and often will opt instead to have clear input and output phases for each command.
 This half-duplex signaling behavior is especially common in devices with also support Dual- and Quad-mode commands, which are always half-duplex.
-The SPI_HOST IP optionally supports both full-duplex and half-duplex commands in standard mode.
+The SPI_HOST IP optionally supports both full-duplex and half-duplex commands in Standard-mode.
```

Comment:
I'm keen on the consistency! Do we have a glossary somewhere? (This would explain the capital and the hyphen)

Reply:
No particular glossary for this term, I just wanted to make this document internally consistent. We capitalize Standard, Dual and Quad throughout, and also commonly use the shorthands Standard-, Dual- and Quad-(mode) when listing the options (e.g. see line 32 just above this diff).